### DISCUSSION TITLE
Running Ripes without X11

### DISCUSSION THREAD

**Discussion by djupdal** [mortbopet/Ripes#345] [Q&A] [answered]
Ripes has a cli mode to enable using it without a GUI.  However, Qt still complains about the lack of a display.  Usually I give "-platform offscreen" to Qt applications that I want to run without a display, but this platform plugin is not available in the Ripes AppImage.  

Is there something I have missed, or should the AppImage release be rebuilt to include the Qt offscreen plugin?

**Comment by djupdal** [comment, marked as answer]
I'll close this discussion now as I have solved it by recompiling and creating a new AppImage myself.  I created the appimage with linuxdeployqt giving the option "-extra-plugins=platforms/libqoffscreen.so".

**Reply by mortbopet** [reply]
Thank you for the follow-up - We _could_ add a separate special CLI release with the argument you described... However, for now, i think i'll still consider Ripes in CLI mode to be an esoteric enough usecase to not warrant this.

However, interest in the CLI mode has been growing recently, and if the users, usecases and CLI-related contributions grow, I think a release would be warranted.

**Reply by djupdal** [reply]
I fully understand that CLI usage of Ripes is not a priority, I guess most users would never need it. 

But it is a very simple addition to just add the Qt offscreen plugin when building the AppImage.  No separate CLI release would be needed as it does not harm GUI operation in any way.  In fact, I don't see any downside of doing this. 

Thanks anyway for your work on Ripes :-)
